mlc_llm/build.py

```diff
@@ -392,13 +394,12 @@ def build(mod_deploy: tvm.IRModule, args: argparse.Namespace) -> None:
     )
 
     debug_dump_shader(ex, f"{args.model}_{args.quantization.name}_{target_kind}", args)
-    lib_path = os.path.join(args.artifact_path, output_filename)
-    ex.export_library(lib_path, **args.export_kwargs)
-    print(f"Finish exporting to {lib_path}")
+    args.lib_path = os.path.join(args.artifact_path, output_filename)
+    ex.export_library(args.lib_path, **args.export_kwargs)
+    print(f"Finish exporting to {args.lib_path}")
 
 
-def main():
-    args = _parse_args()
+def main(args):
```

Comment:
Hi @CharlieRuan0130 , changing the signature of `main` is okay. But please remember to update the outer [build.py](https://github.com/mlc-ai/mlc-llm/blob/ce23bd62a19b79504aa9844d5e557188fd411811/build.py) (mainly for backward compatibility) and the [entry points](https://github.com/mlc-ai/mlc-llm/blob/ce23bd62a19b79504aa9844d5e557188fd411811/setup.py#L45) which allows you calling `python -m mlc_llm.build.main` with a single `mlc_llm_build` in your command lines.

Reply:
This should be fixed now (we got a new `main()`). We have tested the four ways to build a model: 
- `python build.py`
- `python -m mlc_llm.build`
- `mlc_llm_build`
- (through the new API `build_model`)

However, there is a warning when we run `python -m mlc_llm.build`:
```
RuntimeWarning: 'mlc_llm.build' found in sys.modules after import of package 'mlc_llm', but prior to execution of 'mlc_llm.build'; this may result in unpredictable behaviour
```
This is due to how our `__init__.py` does `from .build import build_model`, hence importing `build` twice. Can we leave it like this, or should we make users use the `build_model` API through `from mlc_llm.build import build_model` instead of `from mlc_llm import build_model` (the latter one requires the `__init__.py` change). Thank you!

cc: @sbelcmu

Reply:
I'm thinking about this issue, can we separate the `build_model` and `main` APIs into different files? For example:

1. `build_model()` in `core.py`
2. `main()` still in `build.py`
3. both `__init__.py` and `build.py` should import `core.py`

Reply:
an alternative way is:
1. `build_model()` in `build.py`
2. `main()` in `__main__.py`
3. and `__init__.py` and `__main__.py` imports `build.py`

then user can use:
```
python3 -m mlc_llm ...
```
to build models.

In this case, outermost `build.py` and `setup.py` need to be changed accordingly.

Reply:
Sounds good, looking forward to it!